server/controllers/stock/index.js

```diff
@@ -1169,18 +1173,21 @@ async function listLotsDepotDetailed(req, res, next) {
       db.exec(sqlGetMonthlyStockMovements, [db.bid(params.depot_uuid), params.startDate, params.dateTo]),
     ]);
 
-    data.forEach(current => {
+    const _data = !params.paging ? data : data.rows;
+    const _dataPreviousMonth = !params.paging ? dataPreviousMonth : dataPreviousMonth.rows;
```

Comment:
Again, I suggest calling this `dataPreviousMonthPaged` or something similar to improve code readability.